cmd/nerdctl/build.go

```diff
@@ -78,424 +68,139 @@ If Dockerfile is not present and -f is not specified, it will look for Container
 	return buildCommand
 }
 
-func getBuildkitHost(cmd *cobra.Command, ns string) (string, error) {
-	if cmd.Flags().Changed("buildkit-host") || os.Getenv("BUILDKIT_HOST") != "" {
-		// If address is explicitly specified, use it.
-		buildkitHost, err := cmd.Flags().GetString("buildkit-host")
-		if err != nil {
-			return "", err
-		}
-		if err := buildkitutil.PingBKDaemon(buildkitHost); err != nil {
-			return "", err
-		}
-		return buildkitHost, nil
-	}
-	return buildkitutil.GetBuildkitHost(ns)
-}
-
-func isImageSharable(buildkitHost string, namespace, uuid, snapshotter string, platform []string) (bool, error) {
-	labels, err := buildkitutil.GetWorkerLabels(buildkitHost)
+func buildBuildCommandContext(cmd *cobra.Command, args []string) (*types.BuildCommandContext, error) {
```

Comment:
I suggest changing the name, `buildBuild` sounds duplicated and confusing. maybe `processBuildCmdFlags`